File: 746-min-cost-climbing-stairs/746-min-cost-climbing-stairs.java
class Solution {
    public int minCostClimbingStairs(int[] cost) {
        int n = cost.length;
        int first = cost[0];
        int sec = cost[1];
        if(n <= 2)
            return Math.min(first, sec);
        for(int i=2;i<n;i++){
            int curr = cost[i] + Math.min(first, sec);
            first = sec;
            sec = curr;
        }
        return Math.min(first, sec);      
    }
}

File: 746-min-cost-climbing-stairs/746-min-cost-climbing-stairs.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        dp = dict()
        dp[0] = cost[0]
        dp[1] = cost[1]
        n = len(cost)
        if len(cost) == 2:
            return min(dp[0], dp[1])
        for i in range(2, len(cost)):
            dp[i] = cost[i] + min(dp[i-1], dp[i-2])
        return min(dp[n-1], dp[n-2])
        
        